src/components/BookList.js

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Book from "./Book";
+
+class BookList extends React.Component {
+  static propTypes = {
+    books: PropTypes.array.isRequired,
+    filters: PropTypes.object.isRequired
+  };
+
+  applyFilters = books => {
+    const filters = this.props.filters;
+    for (let filter in filters) {
```

Comment:
Hmmm — nope, `for...in` only allows one argument. But in looking into this, I realized I was doing something kind of silly, and was able to refactor to avoid the problem in the first place. Now `filters` is an array that only contains the currently active filters, rather than an object containing filters with a boolean value.